### PR TITLE
[vdk-plugins] vdk-control-api-auth: Add api-token flow

### PR DESCRIPTION
**Pull request by doks5**
As part of the ongoing work to extract the authentication login of
Versatile Data Kit into a stand-alone library and make it available
to all components (core and plugins) to use, we need to ensure that
the interface surface is as generic as possible.

This change introduces an`Authentication` class, which will act as
the entry point for using the library and authenticating.

Additionally, the api token authentication flow from vdk-control-cli
is also adapted and added to vdk-control-api-auth.

Testing Done: Unit tests.

Signed-off-by: Andon Andonov <andonova@vmware.com>

### CODE REVIEW

File: projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/authentication.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            The URL which exchanges token for access token.
+        :param auth_type:
+            What type of authentication should be used (e.g., refresh_token,
+            basic_auth, etc.).
+        :param cache_locally:
+            A flag, indicating if credentials should be cached locally (in a
+            file).
+        """
+        self._username = username
+        self._password = password
```
Comment (gageorgiev) [inline projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/authentication.py:46]: Are we missing an authentication method? These fields aren't used anywhere; maybe add an empty method to signify that there's code to be added.

File: projects/vdk-plugins/vdk-control-api-auth/README.md [hunk 2]
```diff
@@ -6,7 +6,27 @@ utilities used by vdk-control-cli and other plugins.
...
+```python
+from vdk.plugin.control_api_auth.authentication import Authentication
+
+auth = Authentication(
+    token="<oauth-api-token>",
+    authorization_url="https://some-authorization-endpoint",
+    auth_type="api-token",
+)
+
+auth.authenticate()
```
Comment (tozka) [inline projects/vdk-plugins/vdk-control-api-auth/README.md:28]: How do I get the access token ?

File: projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/authentication.py (hunk 1 above)
Comment (doks5) [reply projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/authentication.py:46]: There will be additional PR-s with the other authentication workflows. I've added a placeholder method.

File: projects/vdk-plugins/vdk-control-api-auth/README.md (hunk 2 above)
Comment (doks5) [reply projects/vdk-plugins/vdk-control-api-auth/README.md:28]: Good question. I've added a method to get the access token, though I wonder if we could just return the access token as part of the `.authenticate` call?

Comment (tozka) [reply projects/vdk-plugins/vdk-control-api-auth/README.md]: It's better to be separate and explicit call (get oauth2 access token). As we'd have another type of authentication mechanism working at teh same time. As part of https://github.com/vmware/versatile-data-kit/pull/809/ I will be adding support for kebreros where authentication is using "Authentication: Negotiate" header. So probably I can add get_kerberos_auth_header method ...

Comment (doks5) [reply projects/vdk-plugins/vdk-control-api-auth/README.md]: I see, ok.

**Review by tozka — APPROVED** [review]

**Pull request merged**